shared/src/simulation/behaviors/transfer.ts

```diff
@@ -0,0 +1,421 @@
+import {
+    IsInt,
+    IsOptional,
+    IsUUID,
+    Min,
+    ValidateNested,
+} from 'class-validator';
+import { groupBy } from 'lodash-es';
+import { Type } from 'class-transformer';
+import {
+    VehicleResource,
+    currentSimulatedRegionOf,
+    getCreate,
+    isInSimulatedRegion,
+    isInSpecificSimulatedRegion,
+} from '../../models/utils';
+import {
+    UUID,
+    cloneDeepMutable,
+    uuid,
+    uuidValidationOptions,
+} from '../../utils';
+import { IsValue } from '../../utils/validators';
+import { addActivity, terminateActivity } from '../activities/utils';
+import { nextUUID } from '../utils/randomness';
+import { getElement } from '../../store/action-reducers/utils';
+import {
+    DelayEventActivityState,
+    LoadVehicleActivityState,
+    RecurringEventActivityState,
+    SendRemoteEventActivityState,
+} from '../activities';
+import { isUnoccupied } from '../../models/utils/occupations/occupation-helpers-mutable';
+import { amountOfResourcesInVehicle } from '../../models/utils/amount-of-resources-in-vehicle';
+import type { ResourceDescription } from '../../models/utils/resource-description';
+import {
+    DoTransferEvent,
+    RequestReceivedEvent,
+    StartTransferEvent,
+    VehiclesSentEvent,
+} from '../events';
+import { LoadOccupation } from '../../models/utils/occupations/load-occupation';
+import { WaitForTransferOccupation } from '../../models/utils/occupations/wait-for-transfer-occupation';
+import { TransferVehicleActivityState } from '../activities/transfer-vehicle';
+import type {
+    SimulationBehavior,
+    SimulationBehaviorState,
+} from './simulation-behavior';
+
+export class TransferBehaviorState implements SimulationBehaviorState {
+    @IsValue('transferBehavior' as const)
+    readonly type = 'transferBehavior';
+
+    @IsUUID(4, uuidValidationOptions)
+    public readonly id: UUID = uuid();
+
+    @IsInt()
+    @Min(0)
+    public readonly loadTimePerPatient: number;
+
+    @IsInt()
+    @Min(0)
+    public readonly personnelLoadTime: number;
+
+    @IsInt()
+    @Min(0)
+    public readonly delayBetweenSends: number;
+
+    @Type(() => StartTransferEvent)
+    @ValidateNested()
+    public readonly startTransferEventQueue: readonly StartTransferEvent[] = [];
+
+    @IsUUID()
```

Comment:
Please include the UUID version (4) and `uuidValidationOptions`.

Reply:
Done